compiler/src/main/kotlin/io/spine/protodata/renderer/InsertionPoint.kt

```diff
@@ -49,7 +64,84 @@ public interface InsertionPoint {
      * @return the line number at which the insertion point should be added.
      * @see LineNumber
      */
-    public fun locate(lines: List<String>): LineNumber
+    @Deprecated("Use locate(Text) instead.")
+    public fun locate(lines: List<String>): LineNumber {
+        val coords = locate(text(lines))
+        return when (coords.specCase) {
+            WHOLE_LINE -> LineNumber.at(coords.wholeLine)
+            INLINE -> {
+                loggerFor(InsertionPoint::class.java)
+                    .atWarning()
+                    .withStackTrace(FULL)
+                    .log("`locate(List<String>)` does not support inline insertion. " +
+                            "A whole line insertion will be generated.")
+                LineNumber.at(coords.inline.line)
+            }
+            END_OF_FILE -> LineNumber.endOfFile()
+            NOT_IN_FILE -> LineNumber.notInFile()
+            else -> throw IllegalStateException("Unexpected file coordinates `$coords`.")
+        }
+    }
+
+    /**
+     * Locates the site where the insertion point should be added.
+     *
+     * An insertion point should only appear once in a file.
+     *
+     * @param text the existing code
+     * @return the coordinates in the file where the insertion point should be added
+     * @see SourceFile.at
+     * @see SourceFile.atInline
+     */
+    public fun locate(text: Text): FileCoordinates
+}
+
+/**
+ * A factory of [FileCoordinates] instances.
```

Comment:
Maybe, it could be a Kotlin extension to `FileCoordinates.Builder`?

E.g.
```kotlin
FileCoordinates.newBuilder().at(42, 0)
```
Then, users won't have to remember that there is one more type (i.e. `CoordinatesFactory`) producing the instances.

Reply:
`CoordinatesFactory` is a trait added to `InsertionPoint`s. It's not meant to be instantiated by the user themselves. I've added the description to reflect that in the doc.